yaksh/static/yaksh/js/chat.js

```diff
@@ -0,0 +1,166 @@
+$(document).ready(function() {
+    if ($("#chat_status").val() == "True" && $("#quiz_status").val() == "False"){
+        $("#open_chat_icon").toggle();
+    }
+    var sender = $('#handle').val();
+    var chatsock;
+    var label;
+    var timer;
+    var page_title = document.title;
+    var course_id = $("#course_id").val();
+
+    $("#main").on("click", function(){
+        // Open chat window
+        var room_url = window.location.protocol + "//" +
+                    window.location.host + "/exam/new/chat/" + course_id;
```

Comment:
I am not sure if the <code>room_url</code> should be a part of the client side. Or at least it should encrypted in some form. Exposing the url to the client can increase vulnerability.

Reply:
I feel it is okay. Just need to make sure from views that no intruder can get messages. I feel I covered checks for user to get messages in views function